### ISSUE TITLE
leftJoinAndSelect with sub-query

### ISSUE THREAD

**Issue by SonderRill** [mikro-orm/mikro-orm#4429]
Hi my question is related to qb **leftJoinAndSelect**. It returns all the associated data. But I need to use a subquery, something like this. 

```
JOIN (
  SELECT *
  FROM other_table
  ORDER BY id DESC
  LIMIT 1
)
```

the table that leftJoinAndSelect joins is too large and I just need to get one record from there, but leftJoinAndSelect joins everything at once. How can I do this with queryBuilder?

leftJoinAndSelect also has a third parameter, I thought maybe this is what me need. But it’s not at all clear what it does, it’s not described in the documentation, and when I applied it, the result is just very strange

**Comment by B4nan** [comment]
You can use the last - 4th, not 3rd - parameter for partial loading:

https://github.com/mikro-orm/mikro-orm/blob/master/tests/features/partial-loading/partial-loading.mysql.test.ts#L97-L112

The third one is the join conditions object. 

This won't use a subquery, not sure why you would want that?

**Comment by giancarlo-dm** [comment]
Sometimes we want to perform a conditional join by not only comparing the ids, but also other columns. A subquery could work on this scenario and even be more flexible. 

So I would also consider support for specifying more conditions (at least the root columns of the related table) on the .leftJoin('property', 'alias', { <extra conditions> }), because right now it only allow the $operators and they are compared to the PK of the related table only. Example:  LEFT JOIN product p ON shows.id = p.show_id and p.enabled = true, this is because sometimes we cannot just use this condition on the WHERE part of the query. Of course, if subqueries were supported (and I think it should be because, in my opinion, query builder should be able use almost all SQL capabilities with exception of edge cases), this could be rewritten as using a subquery, TypeORM, for comparison, supports subqueries on joins, so this could be implemented in MikroORM as well.

**Comment by B4nan** [comment]
While I understand what you want, I dont see how it could work when it comes to entity mapping (which the leftJoinAndSelect method is for). We can surely support any query for `qb.execute()`.

**Comment by B4nan** [comment]
Closing as implemented in v6 via #4747
